### PR TITLE
Feature/2970/test migration from main repo

### PR DESCRIPTION
**Pull request by agduncan94**
https://github.com/dockstore/dockstore/issues/2970

Continuing the process of migrating relevant tests to dockstore_cli.

Any tests from dockstore GeneralIT and WorkflowIT that called the client have been moved here. Note this is a work in progress, and more PRs will be required in the future.

### CODE REVIEW

File: dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java [hunk 1]
```diff
@@ -105,6 +113,152 @@ public void testWorkflowLaunchOrNotLaunchBasedOnCredentials() throws IOException
...
+        Workflow refresh = workflowApi.refresh(workflow.getId());
+        Long workflowId = refresh.getId();
+        WorkflowVersion workflowVersion = refresh.getWorkflowVersions().get(0);
+        Long versionId = workflowVersion.getId();
+
+        // Download unpublished workflow version
+        workflowApi.getWorkflowZip(workflowId, versionId);
+        byte[] arbitraryURL = SwaggerUtility.getArbitraryURL("/workflows/" + workflowId + "/zip/" + versionId, new GenericType<byte[]>() {
+        }, webClient);
+        Path write = Files.write(File.createTempFile("temp", "zip").toPath(), arbitraryURL);
```
Comment (coverbeck) [inline dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java]: You may want to extract tempFile to a variable and call deleteOnExit
```
File f = File.createTempFile("temp", "zip");
f.deleteOnExit();
...
```

It won't matter on CI, but they will start accumulating in local builds.

File: dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java [hunk 2]
```diff
@@ -105,6 +113,152 @@ public void testWorkflowLaunchOrNotLaunchBasedOnCredentials() throws IOException
...
+        }
+        assertTrue(thrownException);
+
+        // Download published workflow version
+        Client.main(
+            new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "workflow", "publish", "--entry", toolpath,
+                "--script" });
+        arbitraryURL = SwaggerUtility.getArbitraryURL("/workflows/" + workflowId + "/zip/" + versionId, new GenericType<byte[]>() {
+        }, CommonTestUtilities.getWebClient(false, null, testingPostgres));
+        write = Files.write(File.createTempFile("temp", "zip").toPath(), arbitraryURL);
```
Comment (coverbeck) [inline dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java]: Same as previous comment.

File: dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java [hunk 3]
```diff
@@ -105,6 +113,152 @@ public void testWorkflowLaunchOrNotLaunchBasedOnCredentials() throws IOException
...
+        zipFile = new ZipFile(write.toFile());
+        assertTrue("zip file seems incorrect",
+            zipFile.stream().map(ZipEntry::getName).collect(Collectors.toList()).contains("md5sum/md5sum-workflow.cwl"));
+
+        // download and unzip via CLI
+        Client.main(new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "workflow", "download", "--entry",
+            toolpath + ":" + workflowVersion.getName(), "--script" });
+        zipFile.stream().forEach((ZipEntry entry) -> {
+            if (!(entry).isDirectory()) {
+                File innerFile = new File(System.getProperty("user.dir"), entry.getName());
```
Comment (coverbeck) [inline dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java]: Unless you're running in the home directory, this won't pass locally, will it? It's assuming the contents were downloaded to the home directory.

File: dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/GeneralIT.java [hunk 4]
```diff
@@ -105,4 +145,426 @@ public void testRemoteLaunchWDLNoFile() {
...
+    @Test
+    public void testWDL2JSON() {
+        File sourceFile = new File(ResourceHelpers.resourceFilePath("wdl.wdl"));
+        Client.main(new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "tool", "convert", "wdl2json", "--wdl",
+            sourceFile.getAbsolutePath(), "--script" });
+        // TODO: Test that output is the expected WDL file
+    }
+
+    @Test
+    @Category(ToilCompatibleTest.class)
```
Comment (coverbeck) [inline dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/GeneralIT.java]: Do we need this category still? Not sure what it is doing.

File: dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/GeneralIT.java [hunk 5]
```diff
@@ -105,4 +145,426 @@ public void testRemoteLaunchWDLNoFile() {
...
+        // record entries
+        List<String> collect = new ZipFile(downloadedZip).stream().map(ZipEntry::getName).collect(Collectors.toList());
+        assert (downloadedZip.exists());
+        assert (downloadedZip.delete());
+
+        // download and unzip via CLI while at it
+        Client.main(new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "tool", "download", "--entry",
+            refresh.getToolPath() + ":" + tag.getName(), "--script" });
+        collect.forEach(entry -> {
+            File innerFile = new File(System.getProperty("user.dir"), entry);
```
Comment (coverbeck) [inline dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/GeneralIT.java]: Made a comment about a similar pattern elsewhere

File: dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java (hunk 3 above)
Comment (agduncan94) [reply dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java]: Yes, it does look like that should be the case. It passes locally for me.

File: dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/GeneralIT.java (hunk 4 above)
Comment (agduncan94) [reply dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/GeneralIT.java]: I am unsure, can ask Denis when he is back.

**Pull request merged**